src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+
+    private float radius;
+
+    @Override
+    public double getArea() {
+        return radius * radius * Math.PI;
+    }
+
+    @Override
+    public Figure getRandomFigureSup() {
+        ColorSupplier nextColor = new ColorSupplier();
+        radius = getRandom().nextInt();
+        setColor(nextColor.getRandomColor()); ;
+        setArea(getArea());
+        return this;
+    }
+
+    @Override
+    public Figure getDefaultFigureSup() {
```

Comment:
remove these methods from each figure. Figure shouldn't be concerned with the logic of randomizing, it's just a model, it's called concern separation - let's delegate this responsibility to FigureSupplier class instead

Reply:
i put it there becose it is different in every figure...
i need FiigureSuplier for all figures???

Reply:
Or i must made many methods like getRandomCircle in FigureSupplier ???